src/Network/Http/Auth/Oauth.php

```diff
@@ -54,10 +49,33 @@ public function authentication(Request $request, array $credentials)
         $credentials['method'] = strtoupper($credentials['method']);
         switch ($credentials['method']) {
             case 'HMAC-SHA1':
+                $hasKeys = isset(
+                    $credentials['consumerSecret'],
+                    $credentials['token'],
+                    $credentials['tokenSecret']
+                );
+                if (!$hasKeys) {
+                    return;
+                }
                 $value = $this->_hmacSha1($request, $credentials);
                 break;
 
+            case 'RSA-SHA1':
+                if (!isset($credentials['privateKeyFile'])) {
+                    return;
```

Comment:
But I think this would be inconsistent to the other checks. It is also a bad mistake if the developer forgot the consumerKey. I would then suggest to consistently throw an exception whenever a mandatory credential key is missed.